libs/vkd3d/command.c

```diff
@@ -17334,11 +17391,15 @@ static void STDMETHODCALLTYPE d3d12_command_queue_ExecuteCommandLists(ID3D12Comm
         {
             if (cmd_list->cmd.iterations[iter].vk_init_commands)
             {
+                cmd_cost[j] = VKD3D_COMMAND_COST_HIGH;
+
                 buffer = &buffers[j++];
                 buffer->sType = VK_STRUCTURE_TYPE_COMMAND_BUFFER_SUBMIT_INFO;
                 buffer->commandBuffer = cmd_list->cmd.iterations[iter].vk_init_commands;
             }
 
+            cmd_cost[j] = cmd_list->cmd.estimated_cost;
```

Comment:
Shouldn't the cost be per iteration?

Reply:
Technically yes, but it just seems like extra complexity that doesn't really make a difference in terms of decision making right now anyway.

Reply:
The added complexity seems fairly minimal, this could just change to cmd_list->cmd.iterations[iter].estimated_cost instead, so I think it's worth doing it.

Reply:
Maybe a simpler solution is to consider the full cost of the ID3D12CommandList (sans first init command buffer), and set costs to 0 for everything except the last VkCommandBuffer we add.

Reply:
Then we could have a init cost and total cost for normal commands.

Reply:
Also missing 0 cost in
```
    if (command_queue->vkd3d_queue->barrier_command_buffer)
    {
        /* Append a full GPU barrier between submissions.
         * This command buffer is SIMULTANEOUS_BIT. */
        buffer = &buffers[j++];
        buffer->sType = VK_STRUCTURE_TYPE_COMMAND_BUFFER_SUBMIT_INFO;
        buffer->commandBuffer = command_queue->vkd3d_queue->barrier_command_buffer;
    }
```